dotcom-rendering/src/web/components/TextBlockComponent.tsx

```diff
@@ -77,23 +87,27 @@ const allowsDropCaps = (format: ArticleFormat) => {
 	}
 };
 
-const shouldShowDropCap = ({
-	format,
-	isFirstParagraph,
-	forceDropCap,
-}: {
-	format: ArticleFormat;
-	isFirstParagraph: boolean;
-	forceDropCap?: boolean;
-}): boolean => {
-	if (allowsDropCaps(format)) {
+const shouldShowDropCaps = (
+	html: string,
+	format: ArticleFormat,
+	isFirstParagraph: boolean,
+	forceDropCap?: boolean,
+) => {
+	const validDropCapFormat = isValidFormatForDropCap(format);
+	// We need to strip any markup from our html string so that we accurately measure
+	// the length that the reader will see. Eg. remove link tag html.
+	const isLongEnough = stripHtmlFromString(html).length >= 200;
```

Comment:
Minor point: I think here we're running the HTML regex every time, even when `validDropCapFormat` is `false`? You could choose to early return `false` when `validDropCapFormat` is `false`, avoiding this?

Reply:
Does this feel more | less readable?
```ts
        const validDropCapFormat = isValidFormatForDropCap(format);
	if (!validDropCapFormat) return false;

	// We need to strip any markup from our html string so that we accurately measure
	// the length that the reader will see. Eg. remove link tag html.
	const isLongEnough = stripHtmlFromString(html).length >= 200;
	if (!isLongEnough) return false;

	// When dropcaps are allowed, we always mark the first paragraph as a drop cap
	if (isFirstParagraph) return true;

	// For subsequent blocks of text, we only add a dropcap if a dinkus was inserted
	// prior to it in the article body (Eg: * * *), causing the forceDropCap flag to
	// be set
	if (forceDropCap) return true;

	return false;
```